### ISSUE TITLE
Popup blocked opening the codespace

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#161959]
Testing #161875

Source: insiders.vscode.dev
Target: GH Codespace

The CS was not opened due to popup blocking:

![image](https://user-images.githubusercontent.com/900690/192528115-23d1d9c7-efb6-43d1-88e6-75d6f191bc6e.png)

**Comment by bpasero** [comment]
No this happened because from my user interaction to opening the window, a lot of time passed. Next time I did this, the window opened. I believe browser vendors have different thresholds where they allow a window from opening when it is close to user action. I know Safari is worst here which even made us implement a fallback:

https://github.com/microsoft/vscode/blob/ea5b244abdd8604bc7a3761a7a625b0e87ac5167/src/vs/workbench/browser/window.ts#L160-L194

**Comment by joyceerhl** [comment]
@bpasero should we then always have this fallback logic even when not on Safari?

**Comment by bpasero** [comment]
Yeah why not, we have to validate that our logic works in other browsers 👍
